R/g.impute.R

```diff
@@ -114,49 +114,75 @@ g.impute = function(M, I, params_cleaning = c(), desiredtz = "",
       r4[1:(firstmidnighti - 1)] = 1 #-1 because first midnight 00:00 itself contributes to the first full day
     }
     r4[(lastmidnighti):length(r4)] = 1  #ignore everything after the last midnight
-  } else if (params_cleaning[["strategy"]] == 3) { #select X most active days
+  } else if (params_cleaning[["strategy"]] %in% c(3, 5)) { #select X most active days
     #==========================================
     # Look out for X most active days and use this to define window of interest
-    atest = as.numeric(as.matrix(M$metashort[,2]))
+    if (acc.metric %in% colnames(M$metashort)) {
+      atest = as.numeric(as.matrix(M$metashort[,acc.metric]))
+    } else {
+      acc.metric = grep("timestamp|angle", colnames(M$metashort), 
+                        value = TRUE, invert = TRUE)[1]
+      atest = as.numeric(as.matrix(M$metashort[,acc.metric]))
+    }
     ws3 = M$windowsizes[1]
     ws2 = M$windowsizes[2]
-    r2tempe = rep(r2,each = (ws2/ws3))
-    atest[which(r2tempe == 1)] = 0
-    NDAYS = length(atest) / (12*60*24)
-    pend = round((NDAYS - params_cleaning[["ndayswindow"]]) * 4)
-    if (pend < 1) pend = 1
-    atestlist = rep(0,pend)
-    for (ati in 1:pend) { #40 x quarter a day
-      p0 = (((ati - 1)*12*60*6) + 1)
-      p1 = (ati + (params_cleaning[["ndayswindow"]]*4))*12*60*6  #ndayswindow x quarter of a day = 1 week
-      if (p0 > length(atest)) p0 = length(atest)
-      if (p1 > length(atest)) p1 = length(atest)
-      if ((p1 - p0) > 1000) {
+    ws = M$windowsizes[3]
+    r2tempe = rep(r2, each = (ws2/ws3))
+    r1tempe = rep(r1, each = (ws2/ws3))
+    atest[which(r2tempe == 1 | r1tempe == 1)] = 0
+    if (params_cleaning[["strategy"]] == 3) { 
+      # Select the most active 24-h blocks by a rolling window of windowsizes[3]
+      NDAYS = length(atest) / ((60/ws3)*60*24)
+      pend = round((NDAYS - params_cleaning[["ndayswindow"]]) * (24/(ws/60/60)))
+      if (pend < 1) pend = 1
+      atestlist = rep(0, pend)
+      for (ati in 1:pend) { #40 x quarter a day
+        p0 = (((ati - 1)*60/ws3*60*(ws/60/60)) + 1)
+        p1 = (ati + (params_cleaning[["ndayswindow"]]*(24/(ws/60/60))))*60/ws3*60*(ws/60/60)  #ndayswindow x quarter of a day = 1 week
```

Comment:
`(24/(ws/60/60))))*60/ws3*60*(ws/60/60)` is unnecessarily complex and unreadable.
If I understand well the two `(ws/60/60`) parts can be omitted because they cancel each other out.